dll/shellext/zipfldr/CEnumZipContents.cpp

```diff
@@ -41,18 +41,19 @@ class CEnumZipContents :
         CStringW name;
         bool dir;
         unz_file_info64 info;
-        for (ULONG i = 0; i < celt; ++i)
+        for (ULONG i = 0;; ++i)
```

Comment:
Weird that this would be the only place that would require this change.
Indeed, the `for (i = 0; i < celt; ++i)` construct is used in other places there as well, see https://github.com/reactos/reactos/commit/f283a3f9ae16eb047d60272186fcb911c79dc266

But, could you explain why this change is actually needed? Is it because you need to return something in the `pceltFetched` variable, _**even in case of failure**_ ? Because if it's only needed in case of success, the *pceltFetched thing could have been placed after the for-loop.

Reply:
> _**even in case of failure**_

Because it needs to know how many it returned which might be only a partial number of how many you actually asked for. I'll check the other cases.